pkg/kapp/cmd/app/deploy.go

```diff
@@ -224,6 +224,7 @@ func (o *DeployOptions) Run() error {
 		Description:      "update: " + changeSummary,
 		Namespaces:       nsNames,
 		IgnoreSuccessErr: true,
+		CreateChange:     o.DeployFlags.AppChangesMaxToKeep > 0,
```

Comment:
nit: I would pass the value `AppChangesmaxToKeep` down the tree and then move the condition itself to other functions. I think it might read better given the context 🤔